src/test/datascience/jupyter/jupyterSession.unit.test.ts

```diff
@@ -453,4 +383,41 @@ suite('DataScience - JupyterSession', () => {
             });
         });
     });
+
+    suite('Remote Sessions', () => {
+        let remoteSession: ISessionWithSocket;
+        let remoteKernel: Kernel.IKernelConnection;
+        let remoteSessionInstance: ISessionWithSocket;
+        suite('Switching kernels', () => {
+            setup(async () => {
+                remoteSession = mock<ISessionWithSocket>();
+                remoteKernel = mock(KernelConnection);
+                remoteSessionInstance = instance(remoteSession);
+                remoteSessionInstance.isRemoteSession = false;
+                when(remoteSession.kernel).thenReturn(instance(remoteKernel));
+                when(remoteKernel.registerCommTarget(anything(), anything())).thenReturn();
+                when(sessionManager.startNew(anything(), anything())).thenCall(() => {
+                    return Promise.resolve(instance(remoteSession));
+                });
+
+                const signal = mock<ISignal<ISessionWithSocket, Kernel.Status>>();
+                when(remoteSession.statusChanged).thenReturn(instance(signal));
+
+                await connect('connectToLiveKernel');
+            });
+            test('Restart should restart the new remote kernel', async () => {
```

Comment:
Could you add the test for 'switching' is supposed to kill the old kernel? That's the part I'm worried about. The code you removed from hostJupyterNotebook.

Reply:
We can no longer switch kernels this way, switching kernels happens at a higher level in VS Code level.
I'll check if we have tests for that, if not i'll add it.